xarray/backends/api.py

```diff
@@ -188,15 +188,25 @@ def maybe_decode_store(store, lock=False):
         store = filename_or_obj
     elif isinstance(filename_or_obj, basestring):
         if filename_or_obj.endswith('.gz'):
-            if engine is not None and engine != 'scipy':
-                raise ValueError('can only read gzipped netCDF files with '
-                                 "default engine or engine='scipy'")
-            # if the string ends with .gz, then gunzip and open as netcdf file
+            if is_remote_uri(filename_or_obj):
+                if engine !='netcdf4' and engine !='pydap':
+                   raise ValueError('can only read remote gzipped netCDF files '
+                                     "with engine='netcdf4' or 'pydap'")
+            else:
+                if engine is not None and engine != 'scipy':
+                   raise ValueError('can only read gzipped netCDF files with '
+                                     "default engine, engine='scipy'")
+                # if the string ends with .gz, then gunzip and open as netcdf file
             if sys.version_info[:2] < (2, 7):
                 raise ValueError('reading a gzipped netCDF not '
                                  'supported on Python 2.6')
             try:
```

Comment:
This `try`/`except` clause needs to only wrap the call to `ScipyDataStore` -- otherwise it's catching too many potentially wrong conditions.